### PR TITLE
isotpsend: add new options to generate PDU traffic load

### PR DESCRIPTION
**Pull request by hartkopp**
To generate high load PDU traffic for kernel testing.

Signed-off-by: Oliver Hartkopp <socketcan@hartkopp.net>

### CODE REVIEW

File: isotpsend.c [hunk 1]
```diff
@@ -178,6 +182,22 @@ int main(int argc, char **argv)
 		    }
 		    break;
 
+	    case 'l':
+		    if (optarg[0] == 'i') {
+			    loops = 0; /* infinite loop */
+		    } else {
+			    loops = strtoul(optarg, (char **)NULL, 10);
```
Comment (marckleinebudde) [inline isotpsend.c:189]: `(char **)` cast not needed

File: isotpsend.c [hunk 2]
```diff
@@ -178,6 +182,22 @@ int main(int argc, char **argv)
...
+			    if (!loops) {
+				    fprintf(stderr, "Invalid argument for option -l !\n");
```
Comment (marckleinebudde) [inline isotpsend.c:191]: nitpick: No space in front of the `!`?

File: isotpsend.c [hunk 3]
```diff
@@ -178,6 +182,22 @@ int main(int argc, char **argv)
...
+				    return 1;
+			    }
+		    }
+		    break;
+
+	    case 'g':
+		    usecs = strtoul(optarg, (char **)NULL, 10);
```
Comment (marckleinebudde) [inline isotpsend.c:198]: `(char **)` cast not needed

File: isotpsend.c [hunk 4]
```diff
@@ -87,6 +89,8 @@ int main(int argc, char **argv)
     static struct can_isotp_ll_options llopts;
     int opt;
     extern int optind, opterr, optopt;
+    unsigned int loops = 1; /* one (== no) loop by default */
+    unsigned int usecs = 0; /* wait before sending the PDU */
```
Comment (marckleinebudde) [inline isotpsend.c:93]: My man page says `usleep()` uses a `useconds_t` as argument.

**Review by marckleinebudde — CHANGES_REQUESTED** [review]

File: isotpsend.c (hunk 1 above)
Comment (hartkopp) [reply isotpsend.c:189]: Ugh. I have tons(!) of these casts in the can-utils code. Should I create an additional patch to remove them entirely?

File: isotpsend.c (hunk 3 above)
Comment (hartkopp) [reply isotpsend.c:198]: ack

File: isotpsend.c (hunk 4 above)
Comment (hartkopp) [reply isotpsend.c:93]: Yes and `useconds_t` is `unsigned long`. I will put `useconds_t` here and let `strtoul()` complain if it doesn't fit ;-)

File: isotpsend.c (hunk 2 above)
Comment (hartkopp) [reply isotpsend.c:191]: Yes. This is our common coding style in can-utils.

Comment (hartkopp) [reply isotpsend.c:191]: Ah! The parser stopped at `if (!loops)` :-D

Will fix that.

**Pull request merged**